vaadin-spring/src/main/java/com/vaadin/flow/spring/scopes/VaadinRouteScope.java

```diff
@@ -159,31 +160,32 @@ private void destroy() {
         private Class<?> currentNavigationTarget;
         private List<Class<? extends RouterLayout>> currentLayouts;
 
-        private Map<Class<?>, Set<String>> beanNamesByNavigationComponents = new HashMap<>();
```

Comment:
If I recall the root cause properly, the bug was in preserving the route scoped bean even outside of owner navigation target, when refresh is done in the browser. And that's why the destroy method was not invoked, because the bean itself was not destroyed. From a very 'outer' point of view, something was broken after refresh event, making the beans 'extend' their scope.

From the changes in the current PR, I'm not sure what is the essential point the changed code, which fixes that. 
I clearly see that the `beanNamesByNavigationComponents` map/holder is now moved to `RouteBeanStore` object, which is in turn mapped to UI objects and stored in the user's session.

So, do I understand right, that before this change the UI object was destroyed after refresh and this map was lost. And **this was the root cause of preserving beans**? Now this map is in the session, so browser refresh does not impact it.

Reply:
yes.